src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -1927,16 +1932,17 @@ private static bool IsUserDefinedTrueOrFalse(BoundUnaryOperator @operator)
                 // is right. For example, C# allows the assignment statement:
                 //    (i) = 9;  
                 // So I don't assume this code should special case parenthesized expressions.
-                TypeSymbol type = null, convertedType = null;
+                TypeSymbol type = null, convertedType;
+                NullabilityInfo nullability = boundExpr.TopLevelNullability, convertedNullability;
                 Conversion conversion;
```

Comment:
>Conversion conversion [](start = 16, length = 21)

Consider moving `convertedType`, `convertedNullability`, and `conversion` declarations down to next `if`.